src/shared/components/query/GenesetsVolcanoSelector.tsx

```diff
@@ -0,0 +1,313 @@
+import {Geneset} from "../../api/generated/CBioPortalAPIInternal";
+import {IColumnFormatterData} from "../enhancedReactTable/IColumnFormatter";
+import {IColumnDefMap} from "../enhancedReactTable/IEnhancedReactTableProps";
+import {Td} from 'reactable';
+import {toPrecision} from "../../lib/FormatUtils";
+import { getHierarchyData } from "shared/lib/StoreUtils";
+import ReactSelect from "react-select";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+import { VictoryChart, VictoryScatter, VictoryTheme, VictorySelectionContainer,
+    VictoryAxis, VictoryLabel, VictoryLine } from 'victory';
+
+const styles = styles_any as {
+    GenesetsVolcanoSelectorWindow: string,
+    selectButton: string,
+    selectionColumnHeader: string,
+    selectionColumnCell: string,
+};
+
+class GenesetsVolcanoTable extends EnhancedReactTable<Geneset> {}
+
+export interface GenesetsVolcanoSelectorProps
+{
+    gsvaProfile: string;
+    sampleListId: string|undefined;
+    onSelect: (map_genesets_selected:ObservableMap<boolean>) => void;
+}
+
+@observer
+export default class GenesetsVolcanoSelector extends React.Component<GenesetsVolcanoSelectorProps, {plotData:{x: number, y: number, fill: string}[]}>
+{
+    @observable tableData: Geneset[] = [];
+    @observable allTableData: Geneset[] = [];
+    maxY = 1;
+    plotData = autorun(() => {
+        const newPlotData: {x: number, y: number, fill: string}[] = [];
+        for (const tableDatum of this.allTableData) {
+            const xValue = tableDatum.representativeScore;
+            const yValue = -(Math.log(tableDatum.representativePvalue)/Math.log(10));
+            if (yValue > this.maxY) {
+                this.maxY = yValue;
+            }
+            const fillColor = this.map_genesets_selected.get(tableDatum.name) ? "tomato" : "3786C2";
+            newPlotData.push({x: xValue, y: yValue, fill: fillColor});
+        }
+        this.setState({plotData: newPlotData});
+    });
+    @observable selectedPercentile: {label: string, value: string} = {label: '75%', value: '75'};
+    @observable isLoading = true;
+    @observable percentileHasChanged = false;
+    readonly percentileOptions = [{label: '50%', value: '50'}, {label: '75%', value: '75'}, {label: '100%', value: '100'}];
+    @observable private map_genesets_selected = new ObservableMap<boolean>();
+    constructor(props:GenesetsVolcanoSelectorProps)
+    {
+        super(props);
+        this.state = {plotData: []};
+        this.percentileChange = this.percentileChange.bind(this);
+        this.updateSelectionFromPlot = this.updateSelectionFromPlot.bind(this);
+    }
+    
+    componentDidMount()
+    {
+        this.fetchData();
+    }
+    
+    componentDidUpdate()
+    {
+        if (this.percentileHasChanged) {
+            this.isLoading = true;
+            this.fetchData();
+            this.percentileHasChanged = false;
+        }
+    }
+    
+    private columns:IColumnDefMap = {
```

Comment:
this is probably our fault for not having better documentation but we don't want to use ReactTable or EnhancedReactTable anymore.   I don't want to make you do this work though.   Let me take a look at the functionality in the browser and then get back to you.  sorry for this confusion.